src/test/java/wromaciej/moviesrental/movies/service/CartServiceTest.java

```diff
@@ -48,6 +50,48 @@ public void shouldReturn3p5ForMovie1(){
         assertThat(rentalRate, equalTo(3.5));
     }
 
+    @Test
+    public void shouldReturn0ForNoRentalDays(){
+        // GIVEN
+
+        int rentalDays=0;
+        Mockito.when(movieRepository.findOne(movie1.getId())).thenReturn(movie1);
+
+        // WHEN
+        double rentalRateForDays = cartService.calculateRentalRateForDays(movie1.getId(),rentalDays);
+
+        // THEN
+        assertThat(rentalRateForDays, equalTo(0.0));
+    }
+
+    @Test
+    public void shouldReturn7For2RentalDays(){
+        // GIVEN
+
+        int rentalDays=2;
+        Mockito.when(movieRepository.findOne(movie1.getId())).thenReturn(movie1);
+
+        // WHEN
+        double rentalRateForDays = cartService.calculateRentalRateForDays(movie1.getId(),rentalDays);
+
+        // THEN
+        assertThat(rentalRateForDays, equalTo(7.0));
+    }
+
+    @Rule public final ExpectedException exception = ExpectedException.none();
+    @Test
+    public void shouldThrowExceptionForNegativeRentalDays() throws  IllegalArgumentException{
+        // GIVEN
+        int rentalDays=-1;
+        Mockito.when(movieRepository.findOne(movie1.getId())).thenReturn(movie1);
+        // WHEN
+        exception.expect(IllegalArgumentException.class);
+        //THEN
+        cartService.calculateRentalRateForDays(movie1.getId(),rentalDays);
+
+    }
+
+
```

Comment:
Write a test for a case when movie does not exist - you would get a null from MovieService.